app/models/user.rb

```diff
@@ -4,4 +4,7 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
   has_many :medical_records
+  validates :password, presence: true, length: { minimum: 8 }
```

Comment:
The password is handled by the devise gem, not as a regular attribute.
The reason is that the password is not stored in plain text, so this validation is not actually counting the password, instead, it is counting a string that represents the encrypted password and that is always more than 8 characters.